app/components/emoji_picker/emoji_picker.js

```diff
@@ -0,0 +1,249 @@
+        super(props);
+
+        this.state = {
+            emojis: props.emojis,
+            searchTerm: ''
+        };
+    }
+
+    changeSearchTerm = (text) => this.setState({searchTerm: text})
+
+    cancelSearch = () => this.setState({searchTerm: ''})
+
+    buildData = () => {
+        const {emojis, searchTerm} = this.state;
+
+        const nextEmojis = emojis.map((section) => {
+            const {title, data} = section;
+
+            const nextData = data.reduce((s, emoji) => {
+                if (searchTerm && emoji.includes(searchTerm.toLowerCase())) {
+                    s[0].items.push(emoji);
+                } else if (!searchTerm) {
+                    s[0].items.push(emoji);
+                }
+
+                return s;
+            }, [{
+                key: `${title}-emojis`,
+                items: []
+            }]);
+
+            return {
+                key: title,
+                title,
+                data: nextData
+            };
+        });
+
+        return nextEmojis.filter((emojiSection) => emojiSection.data[0].items.length);
+    }
+
+    renderSectionHeader = ({section}) => {
+        const {theme} = this.props;
+        const styles = getStyleSheetFromTheme(theme);
+
+        return (
+            <View key={section.title}>
+                <Text style={styles.sectionTitle}>{section.title}</Text>
+            </View>
+        );
+    }
+
+    renderEmojis = (emojis, index) => {
+        const {theme} = this.props;
+        const styles = getStyleSheetFromTheme(theme);
+
+        return (
+            <View
+                key={index}
+                style={styles.columnStyle}
+            >
+                {emojis.map((emoji, emojiIndex) => {
+                    const style = [styles.emoji];
+                    if (emojiIndex === 0) {
+                        style.push(styles.emojiLeft);
+                    } else if (emojiIndex === emojis.length - 1) {
+                        style.push(styles.emojiRight);
+                    }
+
+                    return (
+                        <TouchableOpacity
+                            key={emoji}
+                            style={style}
+                            onPress={() => {
+                                this.props.onEmojiPress(emoji);
+                            }}
+                        >
+                            <Emoji
+                                emojiName={emoji}
+                                size={EMOJI_SIZE}
+                            />
+                        </TouchableOpacity>
+                    );
+                })}
+            </View>
+        );
+    }
+
+    renderItem = ({item}) => {
+        const {theme} = this.props;
+        const styles = getStyleSheetFromTheme(theme);
+
+        const numColumns = Number(((deviceWidth - (SECTION_MARGIN * 2)) / (EMOJI_SIZE + (EMOJI_GUTTER * 2))).toFixed(0));
+
+        const slices = item.items.reduce((slice, emoji, emojiIndex) => {
```

Comment:
Wanted to use SectionList here so that we can take advantage of `scrollToLocation` in case we add the section icons. Was kind of a guess since this design is still somewhat rough.